numpy/_core/meson.build

```diff
@@ -1221,6 +1231,20 @@ if use_svml
   endforeach
 endif
 
+unique_hash_so = static_library(
+  'unique_hash',
+  ['src/multiarray/unique.cpp'],
+  c_args: c_args_common,
+  cpp_args: unique_hash_cpp_args,
+  include_directories: [
+    'include',
+    'src/common',
+  ],
+  dependencies: [
+    py_dep,
```

Comment:
oh actually it might be because you're missing a dependency here?

by the way, why is it so important for this to be its own module? you'd probably have an easier time just adding a new thing to the multiarray module....

Reply:
@ngoldbaum you probably know the meson setup well: is there a nice way to override the exception flags for this file?
Before we compile it as it's own module, maybe it is best to just compile `multiarraymodule` with exception support.
(At least my understanding is that @adrinjalali went back to this setup because it allows to keep the multiarray exception compile flags while the use of the hash-map makes it at least inconvenient, maybe impossible, for this file.)

Reply:
The previous setup was that `unique_hash` was a separate Python module, but now it isn't and it _is_ indeed in `multiarraymodule`. However, the file needs to be compiled with different flags than the other files in that module, hence a separate `static_library` (instead of a separate `py.extension_module`) just to generate the `.so` file to then link the rest with it.